### ISSUE TITLE
[Bug] WASM Node.js examples broken

### ISSUE THREAD

**Issue by cycraig** [iotaledger/identity.rs#336]
## Bug description
Most of the WASM Node.js examples throw errors when run on the latest `dev` branch:

- `create_did`: works ✓
- `create_vc`: ❌
- `create_vp`: ❌
- `manipulate_did`: ❌
- `merkle_key`: ❌
- `resolution`: ❌
- `revocation`: ❌

This issue was originally reported on the Discord identity-discussion channel.

## Language version

Using the WASM bindings for Node.js.

- Language: Rust v1.53.0
- Language: Node.js v14.17.3

## IOTA Identity version
Which version of IOTA Identity are you using?

- Version (version number, commit, or branch): `dev` branch (commit 7b07d50646f7c919143ee0d0fbdbe6d0bde328c8)

## Hardware specification
What hardware are you using? (Delete if not relevant)

- Operating system: Windows 10 19043.1149 / Ubuntu 20.04
- RAM: 16GB
- Cores: 12/24
- Device: desktop AMD CPU (shouldn't matter)

## Steps To reproduce the bug

1. Checkout the latest version of `identity.rs` on the `dev` branch (commit 7b07d50646f7c919143ee0d0fbdbe6d0bde328c8 at time of writing)
2. Build the Node.js WASM bindings:
    ```bash
    cd bindings/wasm
    npm install
    npm run build:nodejs
    ```
3. Run any of the broken examples listed above, e.g. `npm run example:node -- create_vc`

## Expected behaviour

All WASM/Node.js examples should run without errors when executing `npm run example:node -- all`

## Actual behaviour

Most Node.js examples throw an error.

## Errors

The errors for all examples are the same, with `passStringToWasm0`/`Document.set previousMessageId` causing the error.

```
 npm run example:node -- create_vc

> @iota/identity-wasm@0.3.4 example:node C:\<snip>\identity.rs\bindings\wasm
> node examples/node/node.js "create_vc"

Identity Creation: https://explorer.iota.org/mainnet/transaction/[object Object]
Identity Creation: https://explorer.iota.org/mainnet/transaction/[object Object]
Err > RuntimeError: memory access out of bounds
    at dlmalloc::dlmalloc::Chunk::size::h477966db562662ed (<anonymous>:wasm-function[4821]:0x307492)
    at __rdl_realloc (<anonymous>:wasm-function[799]:0x1f82f7)
    at __rust_realloc (<anonymous>:wasm-function[4297]:0x3054f9)
    at __wbindgen_realloc (<anonymous>:wasm-function[4099]:0x303cd5)
    at passStringToWasm0 (C:\Work\iota\identity.rs\bindings\wasm\node\identity_wasm.js:89:15)
    at Document.set previousMessageId [as previousMessageId] (C:\<snip>\identity.rs\bindings\wasm\node\identity_wasm.js:1300:20)
    at manipulateIdentity (C:\<snip>\identity.rs\bindings\wasm\examples\node\manipulate_did.js:46:27)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async createVC (C:\<snip>\identity.rs\bindings\wasm\examples\node\create_vc.js:25:20)
    at async main (C:\<snip>\identity.rs\bindings\wasm\examples\node\node.js:29:20)   
```

**Comment by cycraig** [comment]
Git bisect reports 4e4bd0ecc535c370677719b538a1cf13df3916b7 as the first commit where the Node.js examples started breaking. That commit merged in PR #292, which contains a lot of changes so it may take some time to sift through.

**Comment by cycraig** [comment]
It looks like `messageId` was changed from a simple string to be an object.

**Before:**
```
messageId: '5acac02f413a9fdc14499c7b7cbe5861798a0d0c83f3f59cf47df2e8299f298d'
```

**After:**
```
messageId: {
  network: 'main',
  message_id: '5acac02f413a9fdc14499c7b7cbe5861798a0d0c83f3f59cf47df2e8299f298d',
  network_id: 1454675179895816200,
  nonce: 633644
}
```
<br><br>

Thus it suffices to change the lines of code in the examples where the previous message id is set from
```
doc.previousMessageId = messageId;
```
to
```
doc.previousMessageId = messageId.message_id;
```

We'll should probably add convenience methods for this. Also, we probably wouldn't want to require the whole object to be passed to `previousMessageId` since the user may be missing fields like the `network_id`/`nonce` (and they extra fields are mostly irrelevant apart from `network`).
